### PR TITLE
[Profile] Re-revert mem profile on dashboard/agent

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This PR was reverted due to mac failure. This fixes the issue. (it's a combination of revert + https://github.com/ray-project/ray/pull/29427)

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by fishbone — APPROVED** [review]
LGTM as long as mac build passed

**Comment by rkooo567** [comment]
It passed the mac build. I didn't run after merging the master, but I assume it should pass..

**Comment by rickyyx** [comment]
We will need to cherry pick this

**Pull request merged**
